crates/ruma-api-macros/src/api/metadata.rs

```diff
@@ -160,26 +170,50 @@ impl Parse for Metadata {
         }
 
         if let Some(added) = &added {
-            if stable_path.is_none() {
+            if stable_or_r0.is_none() {
                 return Err(syn::Error::new_spanned(
                     added,
-                    "added version is defined, but no stable path exists",
+                    "added version is defined, but no stable or r0 path exists",
+                ));
+            }
+        }
+
+        if let Some(r0) = &r0_path {
+            let added = added.as_ref().expect("we error if r0 or stable is defined without added");
+
+            if added.major == 1.try_into().unwrap() && added.minor > 0 {
+                return Err(syn::Error::new_spanned(
+                    r0,
+                    "r0 defined while added version is newer than v1.0",
+                ));
+            }
+
+            if stable_path.is_none() {
+                return Err(syn::Error::new_spanned(r0, "r0 defined without stable path"));
+            }
+
+            if !r0.0.value().contains("/r0/") {
```

Comment:
Can you add a `value` method to `EndpointPath` to avoid this weird `r0.0.` thingy? ^^